Manatee.Json/Parsing/BoolParser.cs

```diff
@@ -8,102 +8,167 @@ namespace Manatee.Json.Parsing
 {
 	internal class BoolParser : IJsonParser
 	{
+		static readonly string UnexpectedEndOfInput = "Unexpected end of input.";
+
 		public bool Handles(char c)
 		{
-			return c.In('t', 'T', 'f', 'F');
+			switch (c)
+			{
+				case 't':
+				case 'T':
+				case 'f':
+				case 'F':
+					return true;
+			}
+			return false;
 		}
+
 		public string TryParse(string source, ref int index, out JsonValue value, bool allowExtraChars)
 		{
-			char[] buffer;
-			int count;
-			if (source[index].In('t', 'T'))
-			{
-				buffer = new char[4];
-				count = 4;
-			}
-			else
-			{
-				buffer = new char[5];
-				count = 5;
-			}
-			count = Math.Min(count, source.Length - index);
-			for (var i = 0; i < count; i++)
-			{
-				buffer[i] = source[index + i];
-			}
-			var result = new string(buffer).ToLower();
-			if (result == "true")
+			value = null;
+
+			if (source[index] == 't' || source[index] == 'T')
 			{
+				if (index + 4 >= source.Length)
+					return UnexpectedEndOfInput;
+
+				if (source.IndexOf("true", index, 4, StringComparison.OrdinalIgnoreCase) != index)

```

Comment:
JSON specifies that true/false should be lower-case.  I don't mind having an option to ignore case, but I'd like the default to be conformant.

Reply:
I was following the original logic which did a `ToLower()` before comparing to `"true"` or `"false"`.

Reply:
fair enough.  I think I'll change that logic and move it to an option.